Realm/RLMUtil.mm

```diff
@@ -248,6 +249,50 @@ id RLMValidatedObjectForProperty(id obj, RLMProperty *prop, RLMSchema *schema) {
     return outArray;
 };
 
+NSArray *RLMCollectionValueForKey(NSString *key, RLMRealm *realm, RLMObjectSchema *objectSchema, size_t count, size_t (^indexGenerator)(size_t)) {
+    if (count == 0) {
+        return @[];
+    }
+    BOOL keyIsSelf = [key isEqualToString:@"self"];
+    NSMutableArray *results = [NSMutableArray arrayWithCapacity:count];
+    RLMObjectBase *accessor;
+    tightdb::Table *table = nullptr;
+    if (!keyIsSelf) {
```

Comment:
as it is, I don't find it to be very hard to read, but if you _really_ want it separated out, I suppose I could, but I would mean duplicating a lot of the core logic